### PR TITLE
Update documentation and readme

### PR DESCRIPTION
**Pull request by ptersilie**

### CODE REVIEW

**Comment by ltratt** [comment]
There are quite a few test failures?

**Comment by ptersilie** [comment]
Yes, I didn't realise `cargo test` runs the doc examples as well!

**Comment by ptersilie** [comment]
Docs fixed now. Ready to squash if everything looks ok.

File: README.md [hunk 1]
```diff
@@ -1,22 +1,59 @@
...
+Let's take as an example the two-dimensional vector
+```
+1 0 0
+2 0 0
+3 0 0
+0 0 4
+```
+represented as a one dimensional vector `v = [1,0,0,2,0,0,3,0,0,0,0,4]` with row-length 3.
+Storing this vector in memory is wasteful as the majority of its elements is 0. We can compress
+this vector using row displacement, which merges all rows into a vector such that no two
```
Comment (ltratt) [inline README.md]: Can we say something like "such that non-zero entries are never mapped to the same position"? ["No two non-zero" is a bit hard to parse at first.]

File: src/lib.rs [hunk 2]
```diff
@@ -33,6 +85,17 @@ where
     T: AsPrimitive<usize> + FromPrimitive + Ord + PrimInt + ToPrimitive + Unsigned,
     usize: AsPrimitive<T>,
 {
+    /// Constructs a new SparseVec from a vector (with a user-defined backing storage type) where
```
Comment (ltratt) [inline src/lib.rs]: I think this should be "Constructs a new `SparseVec` from a `Vec` of unsigned integers." [The "user-defined backing storage" bit isn't relevant here.]

**Comment by ltratt** [comment]
I'd like to get a reference to the paper into a comment into the code, because that's the most likely way that future maintainers will notice it. Otherwise, I just have a few minor comments.

**Comment by ptersilie** [comment]
Addressed your comments.

**Comment by ltratt** [comment]
Looks good. Please squash.

**Comment by ptersilie** [comment]
Squashed!
